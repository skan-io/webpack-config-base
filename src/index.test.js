import webpackConfigurator from '.';
import {getEntries} from '.';


jest.mock('react-entry-loader/entry', ()=> (
  jest.fn((obj)=> (path)=> ({obj, path}))
));

test('webpack-config regression', ()=> {
  expect(webpackConfigurator(
    ['index.html.js'], 'build/pkg', 8080, 'favicon.png'
  )('production', 'test-url.com', 'path', 'version-test')
  ).toMatchSnapshot();
});

test('webpack-config regression', ()=> {
  expect(webpackConfigurator(
    ['index.html.js'], 'build/pkg', 8080, 'favicon.png', true
  )('webpack-dev', 'test-url.com', 'path', 'version-test')
  ).toMatchSnapshot();
});

describe('Getting webpack entry points', ()=> {
  it('get appropriate entries', ()=> {
    expect(
      getEntries(['index.html.js'], 'test-url.com', 'version-test')
    ).toEqual({
      index: [
        '@babel/polyfill',
        {
          obj: {
            output: 'index.html',
            appUrl: 'test-url.com',
            version: 'version-test'
          },
          path: './src/index.html.js'
        }
      ]
    });
  });

  it('can handle empty array', ()=> {
    expect(
      getEntries([], 'test-url.com', 'version-test')
    ).toEqual({});
  });

  it('will throw if file type is not .html.js', ()=> {
    expect(
      ()=> getEntries(['index.html'], 'test-url.com', 'version-test')
    ).toThrow(new Error('Must be a .html.js file'));

    expect(
      ()=> getEntries(['index.bad.js'], 'test-url.com', 'version-test')
    ).toThrow(new Error('Must be a .html.js file'));
  });
});
