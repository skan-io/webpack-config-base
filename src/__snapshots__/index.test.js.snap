// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webpack-config regression 1`] = `
Object {
  "devServer": Object {
    "contentBase": "./build",
    "historyApiFallback": Object {
      "rewrites": Array [
        Object {
          "from": /\\^\\\\/\\$/,
          "to": "/path/",
        },
      ],
    },
    "host": "0.0.0.0",
    "inline": true,
    "port": 8080,
    "publicPath": "/path/",
    "stats": "minimal",
  },
  "devtool": false,
  "entry": Object {
    "index": Array [
      "@babel/polyfill",
      Object {
        "obj": Object {
          "appUrl": "test-url.com",
          "output": "index.html",
          "version": "version-test",
        },
        "path": "./src/index.html.js",
      },
    ],
  },
  "mode": "production",
  "module": Object {
    "rules": Array [
      Object {
        "test": /\\(\\\\\\.css\\)\\$/,
        "use": Array [
          "/Users/nick.manks/skan/webpack-config-base/node_modules/mini-css-extract-plugin/dist/loader.js",
          Object {
            "loader": "css-loader",
            "options": Object {
              "camelCase": true,
              "importLoaders": 2,
              "localIdentName": "[name]_[local]_[hash:base64:5]",
              "modules": true,
              "sourceMap": false,
            },
          },
          Object {
            "loader": "postcss-loader",
            "options": Object {
              "plugins": [Function],
              "sourceMap": false,
            },
          },
        ],
      },
      Object {
        "loader": "url-loader?limit=1",
        "test": /\\\\\\.\\(png\\|jpg\\|svg\\|ico\\|gif\\)\\$/,
      },
      Object {
        "include": Array [
          "/Users/nick.manks/skan/webpack-config-base/src",
        ],
        "loader": "babel-loader",
        "options": Object {
          "envName": "production",
        },
        "test": /\\\\\\.jsx\\?\\$/,
      },
    ],
  },
  "optimization": Object {
    "minimize": true,
    "runtimeChunk": Object {
      "name": "runtime",
    },
    "splitChunks": Object {
      "cacheGroups": Object {
        "default": false,
        "react": Object {
          "chunks": "all",
          "name": "react",
          "test": /\\[\\\\\\\\\\\\/\\]node_modules\\[\\\\\\\\\\\\/\\]react/,
        },
      },
      "chunks": "all",
      "name": false,
    },
  },
  "output": Object {
    "devtoolModuleFilenameTemplate": [Function],
    "filename": "[name]-[contenthash].js",
    "path": "/Users/nick.manks/skan/webpack-config-base/build/pkg",
  },
  "plugins": Array [
    EntryTransformPlugin {},
    Object {
      "apply": [Function],
    },
    MiniCssExtractPlugin {
      "options": Object {
        "chunkFilename": "[name]-[contenthash].css",
        "filename": "[name].css",
      },
    },
  ],
}
`;

exports[`webpack-config regression 2`] = `
Object {
  "devServer": Object {
    "contentBase": "./build",
    "historyApiFallback": Object {
      "rewrites": Array [
        Object {
          "from": /\\^\\\\/\\$/,
          "to": "/path/",
        },
      ],
    },
    "host": "0.0.0.0",
    "inline": true,
    "port": 8080,
    "publicPath": "/path/",
    "stats": "minimal",
  },
  "devtool": "#cheap-module-source-map",
  "entry": Object {
    "index": Array [
      "@babel/polyfill",
      Object {
        "obj": Object {
          "appUrl": "test-url.com",
          "output": "index.html",
          "version": "version-test",
        },
        "path": "./src/index.html.js",
      },
    ],
  },
  "mode": "development",
  "module": Object {
    "rules": Array [
      Object {
        "test": /\\(\\\\\\.scss\\|\\\\\\.css\\)\\$/,
        "use": Array [
          "/Users/nick.manks/skan/webpack-config-base/node_modules/mini-css-extract-plugin/dist/loader.js",
          Object {
            "loader": "css-loader",
            "options": Object {
              "camelCase": true,
              "importLoaders": 2,
              "localIdentName": "[name]_[local]_[hash:base64:5]",
              "modules": true,
              "sourceMap": true,
            },
          },
          Object {
            "loader": "postcss-loader",
            "options": Object {
              "plugins": [Function],
              "sourceMap": true,
            },
          },
          Object {
            "loader": "sass-loader",
            "options": Object {
              "data": "@import \\"src/theme.scss\\";",
              "includePaths": Array [
                "/Users/nick.manks/skan/webpack-config-base",
                "/Users/nick.manks/skan/webpack-config-base/node_modules",
              ],
              "sourceMap": true,
            },
          },
        ],
      },
      Object {
        "loader": "url-loader?limit=1",
        "test": /\\\\\\.\\(png\\|jpg\\|svg\\|ico\\|gif\\)\\$/,
      },
      Object {
        "include": Array [
          "/Users/nick.manks/skan/webpack-config-base/src",
        ],
        "loader": "babel-loader",
        "options": Object {
          "envName": "webpack-dev",
        },
        "test": /\\\\\\.jsx\\?\\$/,
      },
    ],
  },
  "optimization": Object {
    "minimize": false,
    "runtimeChunk": Object {
      "name": "runtime",
    },
    "splitChunks": Object {
      "cacheGroups": Object {
        "default": false,
        "react": Object {
          "chunks": "all",
          "name": "react",
          "test": /\\[\\\\\\\\\\\\/\\]node_modules\\[\\\\\\\\\\\\/\\]react/,
        },
      },
      "chunks": "all",
      "name": true,
    },
  },
  "output": Object {
    "devtoolModuleFilenameTemplate": [Function],
    "filename": "[name]-[contenthash].js",
    "path": "/Users/nick.manks/skan/webpack-config-base/build/pkg",
  },
  "plugins": Array [
    EntryTransformPlugin {},
    Object {
      "apply": [Function],
    },
    MiniCssExtractPlugin {
      "options": Object {
        "chunkFilename": "[name]-[contenthash].css",
        "filename": "[name].css",
      },
    },
  ],
}
`;
